force-app/main/default/lwc/jobSearch/jobSearch.js

```diff
@@ -0,0 +1,342 @@
+        this.allJobsByPageNumber = {};
+        this.searchComplete = false;
+        this.sortBy = 'updated';
+        this.sortDirection = 'desc';
+    }
+
+    // Imperative callout since I want to explicitly control when it is done    
+    // Fetches jobs to display, whether by making a callout or fetching jobs that have already been saved from a callout
+    async getJobs() { 
+        // Checks if the requested page of the results has already been obtained by the API and stored.
+        try {
+            if (this.allJobsByPageNumber[Number(this.requestedPage)]) {
+                // Loads the jobs already stored and selected jobs for this page
+                this.currentlyDisplayedJobs = this.allJobsByPageNumber[Number(this.requestedPage)];
+                const selectedRowsForPage = this.selectedJobsByPageNumber[Number(this.requestedPage)] || [];
+                this.currentlySelectedJobs = selectedRowsForPage.map(row => row.id);
+            } else {
+                // API call to get the jobs for the requested page, since they haven't been obtained yet
+                let response = await searchJobs({
+                    jobKeywords: this.jobKeywords,
+                    jobLocation: this.jobLocation,
+                    pageNumber: Number(this.requestedPage)
+                });
+                this.allJobsByPageNumber[Number(this.requestedPage)] = response.jobs; // Stores the returned jobs for the page so we don't need to call to get them again
+                this.totalJobs = Number(response.jobCount); // Stores the total job count to help with pagination since the API only returns 100 jobs but does say the total count
+                this.totalPages = Number(response.maxPages); // The number of pages of jobs we'll display in the LWC
+                this.currentlyDisplayedJobs = response.jobs; // Sets the returned jobs to be visible in the UI now
+                this.currentlySelectedJobs = []; // Doesn't pre-select the page, since this is the first time it has been shown
+                this.checkIfTableShouldDisplay();
+                this.searchComplete = true;
+                }
+        } catch (error) {
+            this.isLoading = false;
+            this.error = error;
+            this.showError(this.error);
+        } finally {
+            this.handlePageNumberMessage();
+            this.shouldNextLastButtonsBeEnabled();
+            this.shouldBackFirstButtonsBeEnabled();
+            this.currentPage = Number(this.requestedPage);
+            this.requestedPage = null;
+            this.isLoading = false;
+        }
+    }
+
+    // Constructs error message to display in the UI
+    showError(error){
+        this.errorMessage = error.body ? (error.body.message || JSON.stringify(error.body)) : 'An unknown error occurred';
+        this.activeError = true;
+    }
+
+    // Checks whether the data table should be displayed by seeing of any jobs were returned
+    checkIfTableShouldDisplay() {
```

Comment:
This method can be simplified since its just returning a boolean value:
`this.shouldDisplayTable = Object.keys(this.allJobsByPageNumber).length !== 0;`